src/cpu.c

```diff
@@ -814,8 +814,7 @@ static int cpu_read (void)
 
 	if (pnumcpu != numcpu || perfcpu == NULL)
 	{
-		if (perfcpu != NULL)
-			free(perfcpu);
+		free(perfcpu);
```

Comment:
Hm, I'm looked to manual, it's really safe to pass NULL to free(), I don't know about this.

#define sfree(ptr) \
    do { \
        if((ptr) != NULL) { \
            free(ptr); \
        } \
        (ptr) = NULL; \
    } while (0)

Why when we have macro with not-null check?  Possible, to avoid function call, which is more expensive.
Sorry for confusion.